### PR TITLE
consolidate api.yml put/delete/patch with post

### PR DESCRIPTION
**Pull request by kbrock**
Currently, we are not reading the `put/edit` or `patch/edit` portion of `api.yml`. Instead, we use the `post/edit`
Deleting the unused put and patch from api.yml makes it clearer

Currently we are reading `post/delete` for POST to collections and `delete/delete` DELETE to collections.
We are using `post/delete` for POST to resources.

Consolidating the privileges into just `post/delete` which will be used for both POST and DELETE. The hope is that a single entry for api.yml makes the rules simpler and will help make it more consistent.

### CODE REVIEW

**Comment by Fryguy** [comment]
Do we also have to then change the verbs (e.g. `*gpppd`) for each of these?  I'm assuming the verbs still exist, its just the permissions definition comes from post?

---

@agrare would also like your review here.

**Comment by kbrock** [comment]
> Do we also have to then change the verbs (e.g. `*gpppd`) for each of these? I'm assuming the verbs still exist, its just the permissions definition comes from post?

Yes, the external API is the same. The privileges for `PUT` and `PATCH` look at `post/edit` both before and after, so they have not changed in the least. We just deleted the sections from `api.yml` that they were previously not being read.

The `DELETE` is changed. Before some `post/delete` were allowed and `delete/delete` were not, and vice versa. After, they both are both allowed or neither is allowed.

It looks like a few of the discovery methods are not not happy / failing tests.
I'll look into that and see if it is a simple addition or if we need to throw this PR away.

**Comment by kbrock** [comment]
pushed update:

In specs, when looking up privileges, it sometimes looked for delete/delete, put/edit, patch/edit.
I could have changed a few hundred lines of specs but opted to modify the helper to ignore the http method parameter instead.
Holler if you want me to modify the specs to fix it. (easy to do but adds a lot of churn and may not be as backport friendly)

Some specs discovered the methods available, and delete/delete is no longer returned. This is a change to the API. I opted to modify the specs but can look into modifying the spec description generator instead.

File: spec/support/api/helpers.rb [hunk 1]
```diff
@@ -50,6 +50,8 @@ def entity_action_identifier(type, entity, action, method = :post)
         end
 
         def action_identifier(type, action, selection = :resource_actions, method = :post)
+          method = action == :read ? :get : :post
```
Comment (Fryguy) [inline spec/support/api/helpers.rb:53]: Does this effectively ignore the passed in `method` then?

Comment (Fryguy) [reply spec/support/api/helpers.rb:53]: Oh I misread your comment, so yes it does ignore it.  In that case, let's maybe change the method signature to say:

```diff
-         def action_identifier(type, action, selection = :resource_actions, method = :post)
+         def action_identifier(type, action, selection = :resource_actions, _method = :post)
```

Comment (kbrock) [reply spec/support/api/helpers.rb:53]: yea. I debated fixing all the callers or here.
Would like your thoughts on whether it is better to "do it right" and have a lot of change/harder to back port or do it quick.

Comment (kbrock) [reply spec/support/api/helpers.rb:53]: heh. your second comment just showed up. just pushed that change.

**Comment by kbrock** [comment]
push:
- change tenant_group spec to pass symbols
- changed helper signature to signify that the `_method` is ignored.

File: config/api.yml [hunk 2]
```diff
@@ -68,15 +65,6 @@
         :identifier: miq_action_edit
       - :name: delete
         :identifier: miq_action_delete
-      :patch:
-      - :name: edit
-        :identifier: miq_action_edit
-      :put:
-      - :name: edit
-        :identifier: miq_action_edit
```
Comment (agrare) [inline config/api.yml:76]: @kbrock are these product features still used when you POST with an action of edit?

Comment (kbrock) [reply config/api.yml:76]: we have never used put/edit, or patch/edit, we always use post/edit.
For all these examples, there should remain a post/edit. If not, then I introduced a bug

Comment (kbrock) [reply config/api.yml:76]: let me try again

Yes. We use `miq_action_edit` when we `PUT/edit` and `POST/edit`.
We are currently, and have always, used `POST/edit` privileges for both.

**Review by agrare — APPROVED** [review]
